### DISCUSSION TITLE
Aggregate multiple batteries with Node-RED

### DISCUSSION THREAD

**Discussion by mr-manuel** [mr-manuel/venus-os_dbus-mqtt-battery#10] [General]
This discussion should be about how aggregating multiple batteries with Node-RED including screenshits and flows.

@pos-ei-don @xury77 may you have an example to post?

**Comment by pos-ei-don** [comment]
I am currently out of the country and cannot do this on my cellphone.
My Flow is currently very basic, but it works very well. I am very happy with this plugin!

**Reply by mr-manuel** [reply]
Could you post it when you are back?

**Reply by xury77** [reply]
I have to do some tests.  In particular, I'm curious about how it works with ESS , but I promise to show my flows and screenshots after Easter time.

**Comment by xury77** [comment]
I have two batteries connected but not really aggregated
So real aggregations must be something like this: https://github.com/Dr-Gigavolt/dbus-aggregate-batteries/blob/main/aggregatebatteries.py
Now I use Lynx as workaround. Perhaps is be possible to use above method for true aggregate

**Comment by teefixx** [comment]
I have two batteries connected via Louisvdw serialbattery driver and aggregate them with mr-manuel MQTT battery.

The values are collected via node red and published mostly to global variables. With that concept I have the values available all over nodered. (Flow: Batterien)

After that the values for the MQTT battery are calculated and published as well. A simple function node converts all neede values to a JSON object and publishes them via MQTT (Flow: virtuelle Batterie)

flows are attached as .txt files

Thank you for this great and versatile driver!


<img width="856" alt="virtuelle Batterie" src="https://user-images.githubusercontent.com/35899354/233777890-363fa71f-f252-462e-a65a-fde4e2936867.png">
<img width="413" alt="Batterien" src="https://user-images.githubusercontent.com/35899354/233777891-e344cadb-ed0a-4313-a251-198fe3773292.png">

[Batterien.json.txt](https://github.com/mr-manuel/venus-os_dbus-mqtt-battery/files/11301072/Batterien.json.txt)
[virtuelle Batterie.json.txt](https://github.com/mr-manuel/venus-os_dbus-mqtt-battery/files/11301073/virtuelle.Batterie.json.txt)

**Reply by mr-manuel** [reply]
Thanks
